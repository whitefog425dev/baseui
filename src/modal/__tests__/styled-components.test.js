/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow
import React from 'react';
import {shallow} from 'enzyme';
import {
  StyledRoot,
  StyledBackdrop,
  StyledDialogContainer,
  StyledDialog,
  StyledClose,
  ModalHeader,
  ModalFooter,
  ModalBody,
  SIZE,
} from '../index.js';

describe('Modal styled components', () => {
  // Basic tests with static styles first
  [
    ['StyledDialogContainer', StyledDialogContainer],
    ['StyledClose', StyledClose],
    ['ModalHeader', ModalHeader],
    ['ModalBody', ModalBody],
    ['ModalFooter', ModalFooter],
  ].forEach(testCase => {
    test(`${testCase[0]} - basic render`, () => {
      const TestCaseClass = testCase[1];
      const component = shallow(<TestCaseClass />);
      expect(component.instance().getStyles()).toMatchSnapshot(
        `${testCase[0]} has correct styles`,
      );
    });
  });

  test('StyledRoot - basic render', () => {
    const component = shallow(<StyledRoot $isOpen={false} />);

    expect(component.instance().getStyles()).toMatchSnapshot(
      'StyledBody has correct styles when closed',
    );

    // Should have pointer-events none when closed
    component.setProps({
      $isOpen: false,
    });
    expect(component.instance().getStyles().pointerEvents).toBe('none');
  });

  test('StyledBackdrop - basic render', () => {
    const component = shallow(
      <StyledBackdrop $isOpen={false} $isVisible={false} />,
    );

    expect(component.instance().getStyles()).toMatchSnapshot(
      'StyledBody has correct styles when closed',
    );

    component.setProps({$isOpen: true});
    expect(component.instance().getStyles().opacity).toBe(0);

    component.setProps({$isVisible: true});
    expect(component.instance().getStyles().opacity).toBe(1);

    component.setProps({$isOpen: false});
    expect(component.instance().getStyles().opacity).toBe(0);

    component.setProps({$animate: false});
    expect(component.instance().getStyles().transitionProperty).toBeUndefined();
  });

  test('StyledDialog - basic render', () => {
    const component = shallow(
      <StyledDialog $isOpen={true} $isVisible={true} />,
    );

    expect(component.instance().getStyles()).toMatchSnapshot(
      'StyledDialog has correct styles',
    );
  });

  test('StyledDialog - full width', () => {
    const component = shallow(
      <StyledDialog $isOpen={true} $isVisible={true} $size={SIZE.full} />,
    );

    const styles = component.instance().getStyles();
    expect(styles.maxWidth).toBe('100%');
    expect(styles.width).toBe('100%');
    expect(styles.alignSelf).toBe('stretch');
  });

  test('StyledDialog - auto width', () => {
    const component = shallow(
      <StyledDialog $isOpen={true} $isVisible={true} $size={SIZE.auto} />,
    );

    const styles = component.instance().getStyles();
    expect(styles.maxWidth).toBe('100%');
    expect(styles.width).toBe('auto');
    expect(styles.alignSelf).toBeUndefined();
  });

  test('StyledDialog - custom width', () => {
    const component = shallow(
      <StyledDialog $isOpen={true} $isVisible={true} $size="800px" />,
    );

    const styles = component.instance().getStyles();
    expect(styles.maxWidth).toBe('100%');
    expect(styles.width).toBe('800px');
    expect(styles.alignSelf).toBeUndefined();
  });
});
