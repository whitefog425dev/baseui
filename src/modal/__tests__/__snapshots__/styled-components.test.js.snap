// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal styled components ModalBody - basic render: ModalBody has correct styles 1`] = `
Object {
  "color": "$theme.colors.foregroundAlt",
  "fontFamily": "$theme.typography.font300.fontFamily",
  "fontSize": "$theme.typography.font300.fontSize",
  "fontWeight": "$theme.typography.font300.fontWeight",
  "lineHeight": "$theme.typography.font300.lineHeight",
  "marginBottom": "$theme.sizing.scale700",
  "marginLeft": "$theme.sizing.scale800",
  "marginRight": "$theme.sizing.scale800",
  "marginTop": "$theme.sizing.scale600",
}
`;

exports[`Modal styled components ModalFooter - basic render: ModalFooter has correct styles 1`] = `
Object {
  "borderTopColor": "$theme.colors.mono400",
  "borderTopStyle": "solid",
  "borderTopWidth": "1px",
  "fontFamily": "$theme.typography.font300.fontFamily",
  "fontSize": "$theme.typography.font300.fontSize",
  "fontWeight": "$theme.typography.font300.fontWeight",
  "lineHeight": "$theme.typography.font300.lineHeight",
  "marginLeft": "$theme.sizing.scale800",
  "marginRight": "$theme.sizing.scale800",
  "marginTop": "$theme.sizing.scale700",
  "paddingBottom": "$theme.sizing.scale500",
  "paddingTop": "$theme.sizing.scale500",
  "textAlign": "right",
}
`;

exports[`Modal styled components ModalHeader - basic render: ModalHeader has correct styles 1`] = `
Object {
  "color": "$theme.colors.foreground",
  "fontFamily": "$theme.typography.font500.fontFamily",
  "fontSize": "$theme.typography.font500.fontSize",
  "fontWeight": "$theme.typography.font500.fontWeight",
  "lineHeight": "$theme.typography.font500.lineHeight",
  "marginBottom": "$theme.sizing.scale600",
  "marginLeft": "$theme.sizing.scale800",
  "marginRight": "$theme.sizing.scale900",
  "marginTop": "$theme.sizing.scale900",
}
`;

exports[`Modal styled components StyledBackdrop - basic render: StyledBody has correct styles when closed 1`] = `
Object {
  "WebkitTapHighlightColor": "transparent",
  "backgroundColor": "rgba(0, 0, 0, 0.5)",
  "bottom": 0,
  "left": 0,
  "opacity": 0,
  "position": "fixed",
  "right": 0,
  "top": 0,
  "touchAction": "none",
  "zIndex": -1,
}
`;

exports[`Modal styled components StyledClose - basic render: StyledClose has correct styles 1`] = `
Object {
  ":hover": Object {
    "color": "$theme.colors.mono800",
  },
  "alignItems": "center",
  "background": "transparent",
  "border": "0",
  "color": "$theme.colors.mono700",
  "cursor": "pointer",
  "display": "flex",
  "height": "$theme.sizing.scale800",
  "justifyContent": "center",
  "outline": 0,
  "padding": 0,
  "position": "absolute",
  "right": "$theme.sizing.scale500",
  "top": "$theme.sizing.scale500",
  "transitionDuration": "$theme.animation.timing100",
  "transitionProperty": "color",
  "width": "$theme.sizing.scale800",
}
`;

exports[`Modal styled components StyledDialog - basic render: StyledDialog has correct styles 1`] = `
Object {
  ":focus": Object {
    "outline": "none",
  },
  "backgroundColor": "$theme.colors.backgroundAlt",
  "borderRadius": "$theme.borders.radius200",
  "margin": "$theme.sizing.scale600",
  "maxWidth": "100%",
  "opacity": 1,
  "pointerEvents": "all",
  "position": "relative",
  "textAlign": "left",
  "transform": "translateY(0)",
  "userSelect": "text",
  "width": null,
}
`;

exports[`Modal styled components StyledDialogContainer - basic render: StyledDialogContainer has correct styles 1`] = `
Object {
  "alignItems": "center",
  "display": "flex",
  "justifyContent": "center",
  "minHeight": "100%",
  "pointerEvents": "none",
  "userSelect": "none",
  "width": "100%",
}
`;

exports[`Modal styled components StyledRoot - basic render: StyledBody has correct styles when closed 1`] = `
Object {
  "bottom": 0,
  "left": 0,
  "overflow": "auto",
  "pointerEvents": "none",
  "position": "fixed",
  "right": 0,
  "top": 0,
  "zIndex": "$theme.zIndex.modal",
}
`;
